crates/host/src/wasmbus/handler.rs

```diff
@@ -456,6 +460,644 @@ impl Messaging for Handler {
+    }
+}
+
+impl Messaging0_3 for Handler {
+    async fn connect(
+        &self,
+        name: String,
+    ) -> anyhow::Result<
+        Result<Box<dyn MessagingClient0_3 + Send + Sync>, messaging0_3_0::types::Error>,
+    > {
+        Ok(Ok(Box::new(MessagingClient {
+            name: name.into_boxed_str(),
+        })))
+    }
+
+    async fn send(
+        &self,
+        client: &(dyn MessagingClient0_3 + Send + Sync),
+        topic: messaging0_3_0::types::Topic,
+        message: messaging0_3_0::types::Message,
+    ) -> anyhow::Result<Result<(), messaging0_3_0::types::Error>> {
+        use wasmcloud_runtime::capability::wrpc::wasmcloud::messaging0_2_0 as messaging;
+
+        let MessagingClient { name } = client
+            .as_any()
+            .downcast_ref()
+            .context("unknown client type")?;
+        {
+            let targets = self.targets.read().await;
+            let target = targets
+                .get("wasmcloud:messaging/producer")
+                .map(AsRef::as_ref)
+                .unwrap_or("default");
+            let name = if name.is_empty() {
+                "default"
+            } else {
+                name.as_ref()
+            };
+            if name != target {
+                return Ok(Err(messaging0_3_0::types::Error::Other(format!(
+                    "mismatch between link name and client connection name, `{name}` != `{target}`"
+                ))));
+            }
+            if let Some(nats) = self.messaging_links.read().await.get(target) {
+                match match message {
+                    messaging0_3_0::types::Message::Host(message) => {
+                        let message = message
+                            .into_any()
+                            .downcast::<Message>()
+                            .map_err(|_| anyhow!("unknown message type"))?;
+                        match *message {
+                            Message::Nats(async_nats::Message {
+                                payload,
+                                headers: Some(headers),
+                                ..
+                            }) => nats.publish_with_headers(topic, headers, payload).await,
+                            Message::Nats(async_nats::Message { payload, .. }) => {
+                                nats.publish(topic, payload).await
+                            }
+                        }
+                    }
+                    messaging0_3_0::types::Message::Wrpc(messaging::types::BrokerMessage {
+                        body,
+                        ..
+                    }) => nats.publish(topic, body).await,
+                    messaging0_3_0::types::Message::Guest(MessagingGuestMessage0_3 {
+                        content_type,
+                        data,
+                        metadata,
+                    }) => {
+                        if content_type.is_some() {
+                            return Ok(Err(messaging0_3_0::types::Error::Other(
```

Comment:
I feel like this should just warn rather than erroring out. It isn't a problem if they set content type. The other thing we could do is just transform it into a header for them. I'd be good with either just warning or trying to be smart